docs/devcontainerjson-reference.md

```diff
@@ -0,0 +1,172 @@
+
+Using `runArgs` via a typical command line, you'll need single quotes if the shell runs into parameters with spaces. However, these single quotes aren't passed on to the executable. Thus, in your `devcontainer.json`, you'd follow the array format and leave out the single quotes:
+
+```json
+"runArgs": ["--device-cgroup-rule=my rule here"]
+```
+
+Rather than:
+
+```json
+"runArgs": ["--device-cgroup-rule='my rule here'"]
+```
+
+We can compare the string and the array versions of `postAttachCommand` as well. You can use the following string format, which will remove the single quotes as part of the shell's parsing:
+
+```json
+"postAttachCommand": "echo foo='bar'"
+```
+
+By contrast, the array format will keep the single quotes and write them to standard out (you can see the output in the dev container log):
+
+```json
+"postAttachCommand": ["echo", "foo='bar'"]
+```
+
+## Variables in devcontainer.json
+
+Variables can be referenced in certain string values in `devcontainer.json` in the following format: **${variableName}**. The following is a list of available variables you can use.
+
+| Variable | Properties | Description |
+|----------|---------|----------------------|
+| `${localEnv:VARIABLE_NAME}` | Any | Value of an environment variable on the **host machine** (in this case, called `VARIABLE_NAME`). Unset variables are left blank. To for example, this would set a variable to your local home folder on Linux / macOS or the user folder on Windows:<br/> `"remoteEnv": { "LOCAL_USER_PATH": "${localEnv:HOME}${localEnv:USERPROFILE}" }` <br /><br /> ⚠️ For Codespaces, the host is in the cloud rather than your local machine.|
+| `${containerEnv:VARIABLE_NAME}` | `remoteEnv` | Value of an existing environment variable inside the container once it is up and running (in this case, called `VARIABLE_NAME`). For example:<br /> `"remoteEnv": { "PATH": "${containerEnv:PATH}:/some/other/path" }` |
+| `${localWorkspaceFolder}`  | Any | Path of the local folder that was opened in the `devcontainer.json` supporting service / tool (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported when using Clone Repository in Container Volume. |
+| `${containerWorkspaceFolder}` | Any | The path that the workspaces files can be found in the container. |
+| `${localWorkspaceFolderBasename}` | Any | Name of the local folder that was opened in the `devcontainer.json` supporting service / tool (that contains `.devcontainer/devcontainer.json`).<br /><br />⚠️ Not yet supported when using Clone Repository in Container Volume. |
+| `${containerWorkspaceFolderBasename}` | Any | Name of the folder where the workspace files can be found in the container. |
+
+## Attached container configuration reference
```

Comment:
Currently this is specific to VS Code and I don't think would be in the reference implementation, correct @chrmarti? If so, we could remove it.

That said, we could add support for an `attach` command that could enable this over time.

Reply:
Removed the content here onward. If it's appropriate to keep any of it @chrmarti, just let us know.

Reply:
I agree, the attach case is not part of the spec. The spec is focused on creating a dev container from a devcontainer.json.